tests/acceptance/02_classes/02_functions/usemodule-returns-nonzero.cf

```diff
@@ -0,0 +1,110 @@
+#!/var/cfengine/bin/cf-agent -f-
+body common control
+{
+        inputs => { "../../default.cf.sub" };
+        bundlesequence  => { default("$(this.promise_filename)") };
+        version => "1.0";
+}
+bundle agent init
+{
+  vars:
+      "foo_usemodule" string =>
+"#!/bin/sh
+echo '=mystring=foo99'
+echo '+class_from_classes_usemodule'
+exit 1
+";
+
+      "foo_commands_module" string =>
+      "#!/bin/sh
+echo '=mystring=foo99'
+echo '+class_from_commands_module'
+exit 1
+";
+
+  files:
+      "${sys.workdir}/modules/foo-usemodule"
+        create => 'true',
+        perms => m( 700 ),
+        edit_defaults => empty,
+        edit_line => append_if_no_line( ${foo_usemodule} );
+
+      "${sys.workdir}/modules/foo-commands_module"
+        create => 'true',
+        perms => m( 700 ),
+        edit_defaults => empty,
+        edit_line => append_if_no_line( ${foo_commands_module} );
+}
+
+bundle agent test
+{
+  meta:
+    "description" -> { "CFE-942" }
+      string => "Test that when a module executed by usemodule() returns nonzero, it's not interpreted as successful";
+
+      "test_soft_fail"
+        string => "any",
+        meta => { "CFE-942" },
+        comment => "usemodule seems to return true no matter if the module exists returning 0 or nonzero";
+
+  classes:
+
+      # Since the module exits with non zero, we should not get this class
+
+      "usemodule_expect_no_class_defined_because_return_nonzero"
+        expression => usemodule( "foo-usemodule", "" ),
+        scope => "namespace",
+        if => isexecutable( "$(sys.workdir)/foo-usemodule" );
```

Comment:
So if for some reason `$(sys.workdir)/foo-usemodule` is not executable you won't get `Pass` or `FAIL` at the end. Will that cause FAIL I hope?

Reply:
i hope so also!